### PR TITLE
Added missing import in boxplot_frame_groupby example

### LINKED ISSUE #28903: groupby boxplot example does not work
#### Problem description
Code found here: 
https://github.com/pandas-dev/pandas/blob/v0.25.1/pandas/plotting/_core.py#L424

Example from https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.core.groupby.DataFrameGroupBy.boxplot.html does not work properly. 

#### Code Sample

```python
import itertools
tuples = [t for t in itertools.product(range(1000), range(4))]
index = pd.MultiIndex.from_tuples(tuples, names=['lvl0', 'lvl1'])
data = np.random.randn(len(index),4)
df = pd.DataFrame(data, columns=list('ABCD'), index=index)
grouped = df.groupby(level='lvl1')
boxplot_frame_groupby(grouped)
grouped = df.unstack(level='lvl1').groupby(level=0, axis=1)
boxplot_frame_groupby(grouped, subplots=False)
```

#### Resulting Error 
```python
---------------------------------------------------------------------------
NameError                                 Traceback (most recent call last)
<ipython-input-49-1a28c2f7125d> in <module>
      5 df = pd.DataFrame(data, columns=list('ABCD'), index=index)
      6 grouped = df.groupby(level='lvl1')
----> 7 boxplot_frame_groupby(grouped)
      8 grouped = df.unstack(level='lvl1').groupby(level=0, axis=1)
      9 boxplot_frame_groupby(grouped, subplots=False)

NameError: name 'boxplot_frame_groupby' is not defined
```

### PR DESCRIPTION
**Pull request by gsyqax**
- [X] closes #28903 
- [x] tests added / passed
- [X] passes `black pandas`
- [X] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

**Comment by MarcoGorelli** [comment]
Thanks @gsyqax - did you build this page of the documentation locally to check it works?

**Comment by gsyqax** [comment]
@MarcoGorelli yes, it is working.

I noticed the example also makes use of 'pd' and 'np' alias without importing and aliasing pandas and numpy. I don't think this can lead to confusing, but do you think I should also add those imports?

**Comment by MarcoGorelli** [comment]
That's OK, it fits with the [docstring guide](https://pandas.pydata.org/docs/development/contributing_docstring.html)

> Import required libraries (except numpy and pandas)

**Comment by MarcoGorelli** [comment]
> @MarcoGorelli yes, it is working.

I just tried building it with your changes and I don't see any plot though, do you?

**Comment by gsyqax** [comment]
I'm not seeing any plots either. I thought a plot was not needed since the current version is not showing any either.

I will add the plots and commit again

**Comment by gsyqax** [comment]
@MarcoGorelli I added some plots and made a few minor changes to make everything clearer, hope it's OK. Also removed whitespace as indicated by pep8speaks.

File: pandas/plotting/_core.py [hunk 1]
```diff
@@ -547,17 +547,25 @@ def boxplot_frame_groupby(
...
+
+        >>> import itertools
+        >>> from pandas.plotting import boxplot_frame_groupby
+        >>> tuples = [t for t in itertools.product(range(1000), range(4))]
+        >>> index = pd.MultiIndex.from_tuples(tuples, names=['lvl0', 'lvl1'])
+        >>> data = np.random.randn(len(index),4)
+        >>> df = pd.DataFrame(data, columns=list('ABCD'), index=index)
+        >>> grouped = df.groupby(level='lvl1')
+        >>> boxplot_frame_groupby(grouped, rot=45, fontsize=12)
+        The ``subplots=False`` option shows the boxplots in a single figure.
```
Comment (MarcoGorelli) [inline pandas/plotting/_core.py:563]:
```suggestion

    The ``subplots=False`` option shows the boxplots in a single figure.
```

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
Nice! Thanks, just tried running it and it looks good. Can you just add a space between the two examples, as suggested below?

**Comment by MarcoGorelli** [comment]
Also, I don't think a whatsnew note is necessary for documentation improvements, could you please revert that change?

**Review by MarcoGorelli — APPROVED** [review]
LGTM

have opened #34353 as I'm not sure what to do about the subplots being too close (but it would warrant a separate PR anyway, I don't think there'd be any need to change that here)

**Comment by gsyqax** [comment]
Great! Many thanks @MarcoGorelli for all the help, really appreciate it!

**Comment by jreback** [comment]
cc @datapythonista @jorisvandenbossche @WillAyd

**Comment by charlesdong1991** [comment]
try `boxplot_frame_groupby(grouped, rot=45, fontsize=12, figsize=(8, 10))` for instance? @gsyqax 

it will avoid having overlapped subplots and make doc look better

**Review by jorisvandenbossche — COMMENTED** [review]
IMO we should not advertise `boxplot_frame_groupby` like that directly, but rather as `df.groupby(..).boxplot(..)`

**Comment by WillAyd** [comment]
Slight preference for what @jorisvandenbossche has suggested above but not a blocker so OK either way

**Comment by gsyqax** [comment]
I applied the change to the plot fig size and replace boxplot_frame_groupby by df.groupby.boxplot. I'm keeping the intermediate var 'grouped' to make things easier to read

File: pandas/plotting/_core.py [hunk 2]
```diff
@@ -547,17 +547,26 @@ def boxplot_frame_groupby(
...
-    >>>
-    >>> grouped = df.unstack(level='lvl1').groupby(level=0, axis=1)
-    >>> boxplot_frame_groupby(grouped, subplots=False)
+    You can create boxplots for grouped data and show them as separate subplots:
+
+    .. plot::
+        :context: close-figs
+
+        >>> import itertools
+        >>> from pandas.plotting import boxplot_frame_groupby
```
Comment (TomAugspurger) [inline pandas/plotting/_core.py:556]: @gsyqax can you remove this import.

Comment (TomAugspurger) [inline pandas/plotting/_core.py:556]:
```suggestion
```

**Comment by jreback** [comment]
lgtm. @TomAugspurger

**Comment by TomAugspurger** [comment]
Thanks @gsyqax!

**Pull request merged**